piecedirectory/piecedirectory.go

```diff
@@ -292,12 +294,35 @@ func (ps *PieceDirectory) addIndexForPiece(ctx context.Context, pieceCid cid.Cid
 	if !errors.Is(err, io.EOF) {
 		return fmt.Errorf("generating index for piece %s: %w", pieceCid, err)
 	}
+	concurrency := AddIndexConcurrency
+	// in an unlikely case if there are less than 8 indexes
+	if concurrency > len(recs) {
+		concurrency = len(recs)
+	}
+
+	chunkSize := len(recs) / concurrency
```

Comment:
Chunk size generally refers to block size. Here it is really the `range` of the slice that each worker will process, so probably worth renaming `chunkSize` to something else, for example `rangeLen` or `rangeLength`.